gift-mode.el

```diff
@@ -42,6 +42,50 @@
 (require 'outline)
 (require 'newcomment)
 
+
+(defvar gift-imenu-comments-regexp "\\(\\(^\\s-*//.*$\\)\\|\\(^\\s-*$\\)\\)+")
+
+(defvar gift-imenu-question-regexp "^\\(\\([^{}]\\)+\\){" )
```

Comment:
In order to detect escaped {, I think this should be something like `\\(\\\\{\\|[^{]\\)+` – i.e. any number of either escaped braces or non-open-brace characters.

Reply:
I have tried some variants, my current solution is a little monster like `"^\\(\\([^\\\\}]\\|\\(\\\\~\\)\\|\\(\\\\~\\)\\|\\(\\\\#\\)\\|\\(\\\\{\\)\\|\\(\\\\}\\)\\|\\(\\\\:\\)\\)*?\\){"`